sdks/go/pkg/beam/io/datastoreio/datastore.go

```diff
@@ -186,12 +200,12 @@ type queryFn struct {
 	// Kind is the datastore kind
 	Kind string `json:"kind"`
 	// Type is the name of the global schema type
-	Type string `json:"type"`
+	Type          string `json:"type"`
+	newClientFunc newClientFuncType
```

Comment:
You've missed the part where I say the value should be checked in a `Setup` method on the DoFn.

As written, this will crash on portable runners, because newClientFunc will always be nil in that context. This is why the suggestion was to check if it's nil, and set up the datastore factory function in a `Setup` method on `queryFn`, and simply passing nil in the Read call.